### PR TITLE
13326 - Saving a module will default to .VMOD, an extension to .VEXT

### PR DESCRIPTION
**Pull request by Cattlesquat**
Up until now, if you make a new module, tell it to "save" or "save as", and type in "Name", you get a module named literally "Name", with no extension. 

This fix corrects that, adding ".vmod" to module saves if no fileext has been added, and ".vext" to a module extension missing a fileext.

### CODE REVIEW

File: vassal-app/src/main/java/VASSAL/tools/ArchiveWriter.java [hunk 1]
```diff
@@ -217,19 +242,38 @@ public void addFile(String fileName, byte[] content) {
     }
   }
 
+  /**
+   * Saves the archive, prompting for a name only if none has ever been provide.
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/tools/ArchiveWriter.java:246]: d

File: vassal-app/src/main/java/VASSAL/tools/ArchiveWriter.java [hunk 2]
```diff
@@ -217,19 +242,38 @@ public void addFile(String fileName, byte[] content) {
...
+   * @throws IOException IOException
+   */
   public void save() throws IOException {
     save(false);
   }
 
+  /**
+   * Saves the archive, prompting for a name only if none has ever been provide.
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/tools/ArchiveWriter.java:254]: d

File: vassal-app/src/main/java/VASSAL/tools/ArchiveWriter.java [hunk 3]
```diff
@@ -240,13 +284,31 @@ protected void write(FileArchive fa, boolean notifyModuleManager)
...
-    final String filename = fc.getSelectedFile().getPath();
+    String filename = fc.getSelectedFile().getPath();
+
+    if (!StringUtils.isEmpty(defaultExtension) && (filename.lastIndexOf('.') < 0)) {
+      if (defaultExtension.indexOf('.') >= 0) {
+        filename = filename + defaultExtension;
+      }
+      else {
+        filename = filename + "." + defaultExtension;
+      }
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/tools/ArchiveWriter.java:307]: Would it be simpler to expect the default extension to supply its own `.`, and then not check for it here?

**Review by uckelman — CHANGES_REQUESTED** [review]

File: vassal-app/src/main/java/VASSAL/tools/ArchiveWriter.java (hunk 3 above)
Comment (Cattlesquat) [reply vassal-app/src/main/java/VASSAL/tools/ArchiveWriter.java]: Fair enough. I'm probably overengineering given the small number of places/times this needs to get called.

**Pull request merged**